packages/flare/tests/bin/test_ingest_events.py

```diff
@@ -30,58 +34,54 @@
 from cron_job_ingest_events import save_last_ingested_tenant_id
 
 
-def test_get_collection_value_expect_none() -> None:
-    kvstore = MagicMock()
+def test_get_collection_value_expect_none(kvstore: FakeKVStoreCollections) -> None:
     assert get_collection_value(kvstore=kvstore, key="some_key") is None
 
 
-def test_get_collection_value_expect_result() -> None:
-    kvstore = MagicMock()
-    kvstore.__contains__.side_effect = lambda x: x == KV_COLLECTION_NAME
-    kvstore[KV_COLLECTION_NAME].data.query.return_value = [
-        {
-            "_key": "some_key",
-            "value": "some_value",
-        },
-    ]
+def test_get_collection_value_expect_result(kvstore: FakeKVStoreCollections) -> None:
+    kvstore.create(name=KV_COLLECTION_NAME, fields={})
+    kvstore[KV_COLLECTION_NAME].data.insert(
+        json.dumps({"_key": "some_key", "value": "some_value"})
+    )
 
     assert get_collection_value(kvstore=kvstore, key="some_key") == "some_value"
 
 
-def test_save_collection_value_expect_insert() -> None:
+def test_save_collection_value_expect_insert(kvstore: FakeKVStoreCollections) -> None:
     key = "some_key"
     value = "some_value"
-    kvstore = MagicMock()
     save_collection_value(kvstore=kvstore, key=key, value=value)
-    kvstore[KV_COLLECTION_NAME].data.insert.assert_called_once_with(
-        json.dumps({"_key": key, "value": value})
-    )
+    assert kvstore[KV_COLLECTION_NAME].data.query() == [{"_key": key, "value": value}]
 
 
-def test_save_collection_value_expect_update() -> None:
+def test_save_collection_value_expect_update(kvstore: FakeKVStoreCollections) -> None:
     key = "some_key"
     value = "update_value"
-    kvstore = MagicMock()
-    kvstore.__contains__.side_effect = lambda x: x == KV_COLLECTION_NAME
-    kvstore[KV_COLLECTION_NAME].data.query.return_value = [
-        {
-            "_key": key,
-            "value": "old_value",
-        },
-    ]
-    save_collection_value(kvstore=kvstore, key=key, value=value)
-    kvstore[KV_COLLECTION_NAME].data.update.assert_called_once_with(
-        id=key,
-        data=json.dumps({"value": value}),
+
+    kvstore.create(name=KV_COLLECTION_NAME, fields={})
+    kvstore[KV_COLLECTION_NAME].data.insert(
+        json.dumps(
+            {"_key": key, "value": "old_value"},
+        )
     )
 
+    assert get_collection_value(kvstore=kvstore, key=key) == "old_value"
+    save_collection_value(kvstore=kvstore, key=key, value=value)
+    assert kvstore[KV_COLLECTION_NAME].data.query() == [{"_key": key, "value": value}]
```

Comment:
Same thing here, you could use `get_collection_value` to assert the value

Reply:
Ha! But it's funny that I see I used it to assert the old value in this one. It was the end of the day. I don't know what I was thinking at that point. 

I mean, it's certainly cleaner. Looks better. I dunno... I still prefer to go directly to the source. I just need someone to force me one way or the other 😂